lib/BuildSystem/BuildSystem.cpp

```diff
@@ -524,7 +524,18 @@ class CommandTask : public Task {
   }
 
   virtual void inputsAvailable(BuildEngine& engine) override {
-    command.inputsAvailable(getBuildSystem(engine).getCommandInterface(), this);
+    bool shouldSkip = !getBuildSystem(engine).getCommandInterface()
```

Comment:
I wonder if we should punt this to the background. The problem with executing this code right here is that this callback is called from the core (serial) build engine loop: the build engine expects clients to immediately offload any work which might be slow. However, since this is a client callback, maybe it is safest to not foist that responsibility onto the client, and instead immediately put this in the background.